src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -494,9 +494,9 @@ private boolean addConnection()
          String password = config.getPassword();
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
-         poolElf.setupConnection(connection, connectionTimeout);
+         PoolBagEntry pbe = poolElf.createBagEntry(connection, connectionTimeout, this);
 
-         connectionBag.add(new PoolBagEntry(connection, this));
+         connectionBag.add(pbe);
```

Comment:
This is *less* readable to me.  On the right I can clearly see that the connection is being setup.  With this change, that action becomes a side-effect of creating a bag entry, and is non-obvious to the reader.